src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4271,13 +4575,33 @@ private static void VerifyFunctionPointerSymbol(TypeSymbol type, CallingConventi
             Assert.Equal(expectedConvention, signature.CallingConvention);
 
             Assert.Equal(returnVerifier.RefKind, signature.RefKind);
+            switch (signature.RefKind)
+            {
+                case RefKind.Out:
+                    Assert.True(false, $"Cannot have a return ref kind of {signature.RefKind}");
+                    break;
+
+                case RefKind.RefReadOnly:
```

Comment:
>case RefKind.RefReadOnly: [](start = 16, length = 25)

What are other possible values? Please assert expectations around modifiers for them too.